osf/management/commands/data_storage_usage.py

```diff
@@ -0,0 +1,615 @@
+                preprint_content_type,
+                start,
+                end,
+            ]
+        )
+        if not dry_run:
+            logger.debug('Writing {} to zip.'.format(filename))
+            write_raw_data(cursor=cursor, zip_file=zip_file, filename=filename)
+
+        logger.debug('Gathering preprint summary at {}'.format(datetime.datetime.now()))
+        summary_data = combine_summary_data(summary_data, summarize(
+            sql=ND_PREPRINT_SIZE_SUM_SQL,
+            content_type=preprint_content_type,
+            start=start,
+            end=end,
+            cursor=cursor,
+        ))
+        logger.debug('Gathering regional preprint summary at {}'.format(datetime.datetime.now()))
+        summary_data = combine_summary_data(summary_data, summarize(
+            sql=REGIONAL_PREPRINT_SIZE_SUM_SQL,
+            content_type=preprint_content_type,
+            start=start,
+            end=end,
+            cursor=cursor,
+        ))
+
+    return summary_data
+
+
+def write_summary_data(filename, summary_data, remote_base_folder):
+    header_row = []
+    summary_row = []
+    for key in summary_data:
+        header_row += [key]
+        summary_row += [summary_data[key]]
+    file_path = '{}{}'.format(TEMP_FOLDER, filename)
+    old_remote = requests.get(
+        url=remote_base_folder['files'],
+        auth=bearer_token_auth(DS_METRICS_OSF_TOKEN),
+        params={'filter[name]': filename},
+    ).json()
+    try:
+        old_remote_data = old_remote['data'][0]
+        upload = old_remote_data['links']['upload']
+        params = {'kind': 'file'}
+        old_file_path = old_remote_data['links']['upload']  # Yes, upload is correct here.
+
+        header_skipped = False
+        with open(file_path, 'w') as new_file:
+            writer = csv.writer(new_file, delimiter=',', lineterminator='\n', quoting=csv.QUOTE_ALL)
+            writer.writerow(header_row)
+            with requests.get(
+                    url=old_file_path,
+                    auth=bearer_token_auth(DS_METRICS_OSF_TOKEN),
+                    stream=True,
+            ) as old_file:
+                reader = csv.reader(old_file.iter_lines(), delimiter=',', lineterminator='\n')
+                for row in reader:
+                    if header_skipped:
+                        writer.writerow(row)
+                    header_skipped = True
+            writer.writerow(summary_row)
+
+    except IndexError:
+        upload = remote_base_folder['upload']
+        params = {
+            'kind': 'file',
+            'name': filename,
+        }
+        with open(file_path, 'w') as new_file:
+            writer = csv.writer(new_file, delimiter=',', lineterminator='\n', quoting=csv.QUOTE_ALL)
+            writer.writerow(header_row)
+            writer.writerow(summary_row)
+
+    upload_to_storage(file_path=file_path, upload_url=upload, params=params)
+
+
+def write_raw_data(cursor, zip_file, filename):
+    data_buffer = StringIO()
```

Comment:
Yes to both. Writing each of those files then streaming them into the temp folder would be a workable solution, but I'm pretty sure it would slow things down. And Matt approved 10k rows in memory a couple months back, so _should_ be fine.